compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1774,14 +1774,14 @@ OMR::Z::TreeEvaluator::tableEvaluator(TR::Node * node, TR::CodeGenerator * cg)
       case AddressTable32bit:
          {
          if(!node->chkCannotOverflow())
-           generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CL, node, selectorReg, numBranchTableEntries, TR::InstOpCode::COND_BNL, node->getSecondChild()->getBranchDestination()->getNode()->getLabel(), false, false);       //make sure the case selector is within range of our case constants
+           generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CL, node, selectorReg, static_cast<int64_t>(numBranchTableEntries), TR::InstOpCode::COND_BNL, node->getSecondChild()->getBranchDestination()->getNode()->getLabel(), false, false);       //make sure the case selector is within range of our case constants
```

Comment:
For this particular location it sounds like `selectorReg` is a 32-bit value, given we do an `SLL` right after. This to me implies we should compare and branch against a 32-bit value as well. Is this correct?

Reply:
Similarly below for the relative case.